framework/auth/core.py

```diff
@@ -42,17 +43,25 @@
 logger = logging.getLogger(__name__)
 
 
-# Hide implementation of token generation
-def generate_confirm_token():
-    return security.random_string(30)
-
-
-def generate_claim_token():
-    return security.random_string(30)
-
+# generate verification key
+def generate_verification_key(verification_type=None):
+    """
+    Generate a one-time verification key with an optional expiration time.
+    The type of the verification key determines the expiration time defined in `website.settings.EXPIRATION_TIME_DICT`.
 
-def generate_verification_key():
-    return security.random_string(30)
+    :param verification_type: None, verify, confirm or claim
+    :return: a string or a dictionary
+    """
+    token = security.random_string(30)
+    # v1 with only the token
+    if not verification_type:
+        return token
+    # v2 with a token and the expiration time
+    expires = dt.datetime.utcnow() + dt.timedelta(minutes=settings.EXPIRATION_TIME_DICT[verification_type])
```

Comment:
This will raise a `KeyError` if `verification_type` is its default value. Is this intended?